modules/thermo-cycler/thermo-cycler-arduino/fan.cpp

```diff
@@ -0,0 +1,49 @@
+#include "fan.h"
+#include "high_frequency_pwm.h"
+
+Fan::Fan()
+{}
+
+void Fan::set_percentage(float p)
+{
+  enable();
+  uint8_t val = p * 255.0;
+  if (val < 0)
```

Comment:
`val` is a `uint8_t` so these checks won't do much. Also, there's a good chance the value will be just truncated to 8 bits, not saturated, which might provide surprising results:

```
dev/opentrons ➤ cat test.c
#include <stdio.h>
#include <stdint.h>

int main(int ignore1, char** ignore2) {
  uint8_t i = 1024;
  printf("%d\n", i);
}
dev/opentrons ➤ gcc test.c
test.c:5:15: warning: implicit conversion from 'int' to 'uint8_t'
      (aka 'unsigned char') changes value from 1024 to 0 [-Wconstant-conversion]
  uint8_t i = 1024;
          ~   ^~~~
1 warning generated.
dev/opentrons ➤ ./a.out
0
```

All in all, it's probably better to check the range of `p` as a float before the conversion.